casper/src/main/scala/coop/rchain/casper/util/rholang/RegistrySigGen.scala

```diff
@@ -109,8 +109,8 @@ object Args {
 
     val id =
       unforgeableNameStr
-        .map(Base16.unsafeDecode(_))
-        .getOrElse(RegistrySigGen.generateUnforgeableNameId(keyPair._1, timestamp))
+        .map(Base16.unsafeDecode)
+        .getOrElse(RegistrySigGen.generateUnforgeableNameId(keyPair._2, timestamp))
```

Comment:
What is ._1 and ._2 here?

Reply:
So, are you saying the types won't allow otherwise?

I would be surprised if this change didn't break the RegistrySigGen tool to the extent where it can't produce the right registry url *at all* (vs "you need to provide timestamp, sk, and the uname that's going to be used for registration, after you've fixed timestamp and sk").

TL;DR did you test that you can register a new contract / an old contract under new timestamp + sk + uname, and see the tests pass?

Reply:
Yes, it's a type error otherwise. This is the same bug that I mentioned to you earlier today. I haven't tested it, no. I guess there should be a test or two for `RegistrySigGen`. I'll add some.

Reply:
@KentShikama `._1` is `PrivateKey`, and `._2` is `PublicKey`.